networkx/algorithms/community/divisive.py

```diff
@@ -0,0 +1,194 @@
+import functools
+
+import networkx as nx
+
+__all__ = [
+    "edge_betweenness_partition",
+    "edge_current_flow_betweenness_partition",
+]
+
+
+def edge_betweenness_partition(G, number_of_sets, weight=None):
```

Comment:
What do you think about making weight keyword-only? Not a blocker of course!

```suggestion
def edge_betweenness_partition(G, number_of_sets, *, weight=None):
```